### PR TITLE
Karma: Don't create duplicate entries for names that differ only in lower- and uppercase

### PR DESCRIPTION
**Pull request by salto-edit**
Suppose someone upvoted "LtWorf" for the first time like so

```
<someone> LtWorf++
<LtData> Prendo nota. LtWorf(1)
```

And later someone upvotes "ltworf" because she is too lazy to spell LtWorf's nickname right, then

```
<someone> ltworf++
<LtData> Prendo nota. LtWorf(2)
```

and the updated database will not contain the key "ltworf" but "LtWorf" with an incremented karma counter.

### CODE REVIEW

**Comment by salto-edit** [comment]
Commits bea77ab and 2b79704 can be squashed.

File: modules/karma.py [hunk 1]
```diff
@@ -70,28 +70,29 @@ def sendmsg(sender, recip, text):
...
-            result = "Nah, non credo di volerlo fare"
-        else:
-            karma[nick] = readval(nick) - 1
-            result = "Prendo nota. %s: %d" % (nick, karma[nick])
-        save()
-        return result
+        return vote(text[:-2], -1)
     return None
 
+def vote(nick, delta=1):
```
Comment (ltworf) [inline modules/karma.py:81]: I don't think that a optional parameter is good here.

File: modules/karma.py [hunk 2]
```diff
@@ -70,28 +70,29 @@ def sendmsg(sender, recip, text):
...
+    if nick.lower() == config['nickname'].lower() and delta > 0:
+        result = "Grazie per la tua stima"
+    elif nick.lower() == config['nickname'].lower() and delta <= 0:
+        return "Nah, non credo di volerlo fare"
+    elif nick.lower() == sender.lower() and delta > 0:
```
Comment (ltworf) [inline modules/karma.py]: sender is not defined

File: modules/karma.py (hunk 1 above)
Comment (salto-edit) [inline modules/karma.py:81]: You find the interface confusing? It's not needed, so go ahead and make it a parameter.

File: modules/karma.py (hunk 2 above)
Comment (salto-edit) [inline modules/karma.py]: See follow up commit 1aae5ff.

**Comment by salto-edit** [comment]
Here you go, don't forget to convert the database again. :tada:

File: modules/karma.py [hunk 3]
```diff
@@ -28,28 +29,22 @@ def init():
...
-        return 0
+        return (nickname, 0)
 
 
 def save():
     f = file("%s/karma" % config['files'], "w")
-    for i in karma:
-        f.write("%s#%d" % (i, karma[i]))
-        f.write("\n")
+    json.dump(f)
```
Comment (ltworf) [inline modules/karma.py:47]: you forgot the data that you want to dump here

Comment (salto-edit) [inline modules/karma.py:47]: Sepyt!

**Pull request merged**